File: backend/scratch/debug_admin_users.py
import sys
import os

# Add the parent directory of the current directory to sys.path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app import create_app
from models import User, Profile

def test_admin_users():
    app = create_app()
    with app.app_context():
        try:
            print("Querying users...")
            users = User.objects().order_by('-created_at')
            print(f"Found {len(users)} users.")
            
            user_list = []
            for u in users:
                print(f"User: {u.name} (id: {u.id}), Role: {u.role}")
                profile = Profile.objects(user=u).first()
                print(f"Profile: {profile}")
                if profile:
                    print(f"Global score: {profile.global_score}")
                user_list.append({
                    "id": str(u.id),
                    "name": u.name,
                    "email": u.email,
                    "role": u.role,
                    "branch": u.branch,
                    "year": u.year,
                    "college": u.college,
                    "global_score": profile.global_score if profile else 0
                })
            print("Successfully processed all users!")
            print(f"User list size: {len(user_list)}")
        except Exception as e:
            import traceback
            print(f"ERROR: {e}")
            traceback.print_exc()

if __name__ == "__main__":
    test_admin_users()

File: backend/src/scripts/scratch/debugAdminUsers.js
/**
 * scripts/scratch/debugAdminUsers.js
 */

import dotenv from 'dotenv';
import connectDB from '../../config/db.js';
import { User, Profile } from '../../models/index.js';
import mongoose from 'mongoose';

dotenv.config();

async function main() {
  await connectDB();
  try {
    console.log('Querying users...');
    const users = await User.find().sort({ created_at: -1 });
    console.log(`Found ${users.length} users.`);

    const userList = [];
    for (const u of users) {
      console.log(`User: ${u.name} (id: ${u._id}), Role: ${u.role}`);
      const profile = await Profile.findOne({ user: u._id });
      console.log(`Profile: ${profile ? 'Found' : 'Not found'}`);
      if (profile) {
        console.log(`Global score: ${profile.global_score}`);
      }
      userList.push({
        id: u._id.toString(),
        name: u.name,
        email: u.email,
        role: u.role,
        branch: u.branch,
        year: u.year,
        college: u.college,
        global_score: profile ? profile.global_score : 0,
      });
    }
    console.log('Successfully processed all users!');
    console.log(`User list size: ${userList.length}`);
  } catch (err) {
    console.error('ERROR:', err.message);
  }
  await mongoose.disconnect();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
